### PR TITLE
Enable retention of original names and adds onnx load support for some unary ops

### PR DESCRIPTION
**Pull request by ksaurabh-cadence**
Summary: 
Adds onnx load support for some ops and addresses issue https://github.com/pytorch/glow/issues/4600

Documentation:
1.  Added onnx load support/tests for ceil/floor/neg.
2.  Modify op-names at load time to prevent loss of original names later on.
3.  Test for retention of original names.

[Optional Fixes #issue]

Test Plan:

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

**Comment by ksaurabh-cadence** [comment]
@jfix Made some changes, so this might look different from what was agreed solution in in #4600. With that solution,CPUOperatorTests were failing because the ONNXLoader/Writer tests in the teardown were too strict with name checking. If this looks little too much, I can simply disable the strict name checking in that test by modifying the DebugDescr api **or** make the name retention an optional feature controlled by command line option. Let me know your preference.

**Comment by jfix** [comment]
@jfix71 this is for you ...

File: lib/Graph/Graph.cpp [hunk 1]
```diff
@@ -621,16 +622,18 @@ std::string Module::getPrefix(llvm::StringRef name) {
...
   std::string legalName = legalizeName(name);
   if (stringTable.find(legalName) == stringTable.end()) {
     auto it = updateTable.insert(legalName);
     if (it.second) {
       return it.first->first();
     }
   }
-
-  std::string prefix = Module::getPrefix(legalName);
+  std::string prefix = (originalNames.find(legalName) == originalNames.end())
```
Comment (jfix71) [inline lib/Graph/Graph.cpp:634]: nit: perhaps add a comment here saying that we retain the `"__[0-9]+"` if it was in the original name?

File: include/glow/Graph/Graph.h [hunk 2]
```diff
@@ -101,14 +104,26 @@ class Module final {
...
+                                    llvm::StringSet<> &updateTable,
+                                    llvm::StringSet<> &originalNames);
 
   /// Registers a name as used by some Node in this module.
   void registerNodeName(llvm::StringRef name) {
     // Don't care if it's already in the set.
     usedNodeNames_.insert(name);
   }
 
+  /// Registers a name from the original model, good to be retained.
```
Comment (jfix71) [inline include/glow/Graph/Graph.h:116]: nit: `Registers a \p name`

File: include/glow/Graph/Graph.h [hunk 3]
```diff
@@ -101,14 +104,26 @@ class Module final {
   /// inserted into \p updateTable.
   static llvm::StringRef uniqueName(llvm::StringRef name,
                                     const llvm::StringSet<> &stringTable,
-                                    llvm::StringSet<> &updateTable);
+                                    llvm::StringSet<> &updateTable,
+                                    llvm::StringSet<> &originalNames);
```
Comment (jfix71) [inline include/glow/Graph/Graph.h:108]: nit: `const`

File: include/glow/Graph/Graph.h [hunk 4]
```diff
@@ -101,14 +104,26 @@ class Module final {
...
+  /// Registers a name from the original model, good to be retained.
+  void registerOriginalName(llvm::StringRef name) {
+    // Don't care if it's already in the set.
+    if (name.size()) {
+      originalNames_.insert(name);
+    }
+  }
+
+  /// \returns the pointer to list of original node names, good to be retained;
+  llvm::StringSet<> *getOriginalNames() { return &originalNames_; }
```
Comment (jfix71) [inline include/glow/Graph/Graph.h:125]: nit: const

File: tests/unittests/OnnxImporterTest.cpp [hunk 5]
```diff
@@ -4316,4 +4316,32 @@ TEST_F(OnnxImporterTest, CustomGlowDAGMultiOp) {
...
+  vector<std::string> origNames = {"a__1",  "a__1", "a__3__3", "a__2",
+                                   "a__1_", "a__b", "a"};
+  auto *currNode = (Node *)getSaveNodeFromDest(graphOutputVar);
+  for (int i = 0; i < origNames.size(); i++) {
+    auto *prevNode = currNode->getNthInput(0).getNode();
+    // Make sure original names are retained in the legalized names.
+    EXPECT_EQ(prevNode->getName().find(origNames[i]), 0);
+    currNode = prevNode;
+  }
 }
```
Comment (jfix71) [inline tests/unittests/OnnxImporterTest.cpp:4347]: nit: while you're here can you add a new line to the end of the file?

File: tests/unittests/OnnxImporterTest.cpp [hunk 6]
```diff
@@ -4301,4 +4316,4 @@ TEST_F(OnnxImporterTest, CustomGlowDAGMultiOp) {
 
   EXPECT_TRUE(resultPartitionedT->isBitwiseEqual(*resultUnpartitonedT,
                                                  /* verbose */ true));
-}
+}
```
Comment (jfix71) [inline tests/unittests/OnnxImporterTest.cpp:4347]: nit: prefer to keep line at the end of file.

**Pull request closed without merging**